common/compositor/factory.cpp

```diff
@@ -17,46 +17,106 @@
 #include "factory.h"
 #include "platformdefines.h"
 
-#ifdef USE_GL
 #include "glrenderer.h"
 #include "glsurface.h"
 #include "nativeglresource.h"
-#elif USE_VK
```

Comment:
It wouldn't even build in this case since you will be missing vulkan headers ? as we will be depending on the fact that Mesa would be built in this case. Does Mesa export the headers alwyas irrespective of not being built with vulkan support inculded ?